sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts

```diff
@@ -269,4 +277,362 @@ describe("AssessmentController(e2e)-verifyAssessmentCalculationOrder", () => {
+        applicationStatus: ApplicationStatus.Completed,
+        currentAssessmentInitialValues: {
+          assessmentWorkflowId: "some fake id",
+          studentAssessmentStatus: StudentAssessmentStatus.Completed,
+          assessmentDate: addDays(-5),
+        },
+      },
+    );
+    // Past full-time application with federal and provincial loans and grants.
+    // Loans will be ignored.
+    await saveFakeApplicationDisbursements(
+      db.dataSource,
+      {
+        student,
+        firstDisbursementValues: [
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaLoan,
+            "CSLF",
+            8,
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaGrant,
+            "CSGD",
+            9,
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.CanadaGrant,
+            "CSGP",
+            10,
+          ),
+          createFakeDisbursementValue(DisbursementValueType.BCLoan, "BCSL", 11),
+          createFakeDisbursementValue(
+            DisbursementValueType.BCGrant,
+            "SBSD",
+            12,
+          ),
+          createFakeDisbursementValue(
+            DisbursementValueType.BCGrant,
+            "BCAG",
+            13,
+          ),
+        ],
+        programYear: programYear,
+      },
+      {
+        offeringIntensity: OfferingIntensity.fullTime,
+        applicationStatus: ApplicationStatus.Completed,
+        currentAssessmentInitialValues: {
+          assessmentWorkflowId: "some fake id",
+          studentAssessmentStatus: StudentAssessmentStatus.Completed,
+          assessmentDate: addDays(-4),
+        },
+      },
+    );
+    // Current application having the first assessment already processed.
+    const currentApplication = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student, programYear: programYear },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.InProgress,
+        currentAssessmentInitialValues: {
+          assessmentWorkflowId: "some fake id",
+          studentAssessmentStatus: StudentAssessmentStatus.Completed,
+          assessmentDate: addDays(-3),
+        },
+      },
+    );
+    const firstAssessmentDate =
+      currentApplication.currentAssessment.assessmentDate;
+    // Create the second assessment for the current application with a different assessment date.
+    const secondAssessment = createFakeStudentAssessment(
+      {
+        auditUser: currentApplication.student.user,
+        application: currentApplication,
+        offering: currentApplication.currentAssessment.offering,
+      },
+      {
+        initialValue: {
+          ...currentApplication.currentAssessment,
+          assessmentWorkflowId: "some fake id",
+          studentAssessmentStatus: StudentAssessmentStatus.InProgress,
+          assessmentDate: addDays(-1),
```

Comment:
From the data perspective, the assessmentDate will be null before the calculations happen.
Please look at the data prepared at line 214. 

![image](https://github.com/bcgov/SIMS/assets/54600590/05988ed6-017b-4a27-a8e7-868dbb9fcf61)